src/generator/types.rs

```diff
@@ -187,6 +191,7 @@ impl ExprFolder {
                             return res;
                         }
                     }
+                    _ => {}
```

Comment:
This (and the `_ => {}` below in `fold_unop`) makes me think it'd be better to split up the `Op` enum into `BinOp` and `UnOp` enum.